packages/node_modules/@ciscospark/react-redux-spark/src/component.js

```diff
@@ -27,71 +30,116 @@ const defaultProps = {
+        createSparkJwt(guestToken, this.props).then(this.loadSparkInstance);
       }
       else {
-        createSpark(accessToken, this.props).then(storeAndListen);
+        createSpark(accessToken, this.props).then(this.loadSparkInstance);
       }
     }
     else {
-      SparkComponent.listenToSparkEvents(spark, props);
+      this.listenToSparkEvents(sparkInstance);
     }
   }
 
-  componentWillReceiveProps(nextProps) {
-    const spark = nextProps.spark.get('spark');
-
-    SparkComponent.setupDevice(spark, nextProps);
-  }
-
-  shouldComponentUpdate(nextProps) {
-    const {props} = this;
-
-    return nextProps.spark !== props.spark;
-  }
-
-  static setupDevice(spark, props) {
+  /**
+   * Register the device if the user has been authenticated
+   * and the device is not regusterd yet.
+   * @param {Object} spark
+   */
+  setupDevice(spark) {
     const {
       authenticated,
       registered,
       registerError,
       registering
-    } = props.spark.get('status').toJS();
+    } = this.props.spark.get('status').toJS();
 
     if (authenticated && !registered && !registering && !registerError) {
-      props.registerDevice(spark);
+      this.props.registerDevice(spark);
+    }
+  }
+
+  /**
+   * Listen to any new events and update the sprak instance
+   * status accordingly
+   * @param {Object} spark
+   */
+  listenToSparkEvents(spark) {
+    spark.listenToAndRun(spark, 'change:canAuthorize', () => {
+      this.props.updateSparkStatus({authenticated: spark.canAuthorize});
+    });
+
+    spark.listenToAndRun(spark, 'change:isAuthenticating', () => {
+      this.props.updateSparkStatus({authenticating: spark.isAuthenticating});
+    });
+
+    spark.listenToAndRun(spark.internal.device, 'change:registered', () => {
+      this.props.updateSparkStatus({registered: spark.internal.device.registered});
+    });
+
+    this.setupDevice(spark);
+  }
+
+  /**
+   * Verify for the plugins, store the spark instance
+   * and listen to any events
+   * @param {Object} sparkInstance
+   */
+  loadSparkInstance(sparkInstance) {
+    this.verifyPlugins(sparkInstance);
+    this.props.storeSparkInstance(sparkInstance);
+    this.listenToSparkEvents(sparkInstance);
+  }
+
+  /**
+   * Verify if all the plugins have been injected
+   * properly to the webex SDK instance
+   * @param {Object} instance
+   */
+  verifyPlugins(instance) {
+    for (const plugin of PLUGINS) {
+      // eslint-disable-next-line no-prototype-builtins
+      if (!instance.hasOwnProperty(plugin)) {
+        this.props.addError({
+          id: 'webex-plugins-missing',
+          displayTitle: 'Something\'s not right. Please try again',
+          displaySubtitle: `Webex SDK instance plugin ${plugin} is not injected to the widget properly`,
+          temporary: false
+        });
+      }
+    }
+
+    for (const internalPlugin of INTERNAL_PLUGINS) {
+      // eslint-disable-next-line no-prototype-builtins
```

Comment:
This needs a `// eslint-disable-reason` if you are not going to use the eslint preferred method.